### DISCUSSION TITLE
How to switch to locale de-de

### DISCUSSION THREAD

**Discussion by wita-sec** [OpenVoiceOS/ovos-buildroot#265] [Q&A] [answered]
How to switch locale to de-de

OVOS headless build 20240313 on RPi4B
out of the box OVOS runs with en-us.

Switching to de-de, OVOS understands my German uterance, OVOS returns German text.
Problem: OVOS speaks German text in English which sounds kind of strange.

What i did in more detail:

switched locale to de-de with ovos-config set -k lang
switched system locale to de-de with raspi-config
then restarted ovos.service

(.venv) ovos@berry:~/.local/state/mycroft $ ovos-config get -k lang
Value: de-de, found in /lang
Value: [], found in /secondary_langs
Value: en-us, found in /hotwords/hey_mycroft_pocketsphinx/lang
Value: en-us, found in /hotwords/wake_up/lang
Value: en-us, found in /hotwords/wake_up_vosk/lang
Value: en, found in /hotwords/hey_ziggy/lang
Value: {'detection_module': 'ovos-lang-detect-ngram-lm', 'translation_module':
'ovos-translate-plugin-server'}, found in /language

(.venv) ovos@berry:/tmp/skill-ovos-energyportal/test $ sudo raspi-config
Generating locales (this might take a while)...
  de_DE.UTF-8... done
  en_US.UTF-8... done
Generation complete.

Testing with
--> ovos-say-to and with my
--> ovosskilltester
results in:
OVOS understands the German uterance
OVOS returns the appropriate German answer text

Problem: OVOS speaks German text in English, which sounds kind of funny.

--> (.venv) ovos@berry:/tmp/skill-ovos-energyportal/test $ ovos-say-to "nenne mir netzwerkadresse"
    <I did not understand what OVOS speaker exposes>

--> (.venv) ovos@berry:/tmp/skill-ovos-energyportal/test $ ./ovosskilltester.sh -t -u "nenne mir netzwerkadresse"
going to test uterances of file /tmp/wita_test_uterance_20240322214159.1711140119
uterance delay was set to 6 seconds

20240322.214159 @ Single Uterance
@ Single Uterance 1 uterance: nenne mir netzwerkadresse
@ Single Uterance 1 relpy:
auf eth0 ist meine I.P Adresse 192 Punkt 168 Punkt 2 Punkt 237.
auf wlan0 ist meine I.P Adresse 192 Punkt 168 Punkt 2 Punkt 238.
================

**Comment by PocketMiner82** [comment, marked as answer]
Hi @wita-sec, did you also change your TTS to one that supports German? I've changed the TTS config to use [Piper](https://github.com/OpenVoiceOS/ovos-tts-plugin-piper).

I use the thorsten-high voice (~/.config/mycroft/mycroft.conf):
```json
  "tts": {
    "module": "ovos-tts-plugin-piper",
    "ovos-tts-plugin-piper": {
      "voice": "thorsten-high"
    },
    "sentence_tokenize": true
  }
```

Note that this will probably be a bit slower, as the voice is generated on device.

You can also try [OVOS TTS Plugin Server](https://github.com/OpenVoiceOS/ovos-tts-server-plugin) which uses community hosted piper servers with this config:
```json
  "tts": {
    "module": "ovos-tts-plugin-server",
    "ovos-tts-plugin-server": {
      "voice": "thorsten-high"
    },
    "sentence_tokenize": true
  }
```

**Reply by wita-sec** [reply]
Hey PocketMiner82, 
thank you for your help on this. The config you suggested does the trick.
